src/contact.rs

```diff
@@ -892,20 +892,10 @@ impl Contact {
             .unwrap_or_default() as usize
     }
 
-    pub fn get_origin_by_id(context: &Context, contact_id: u32, ret_blocked: &mut bool) -> Origin {
-        let mut ret = Origin::Unknown;
-        *ret_blocked = false;
-
-        if let Ok(contact) = Contact::load_from_db(context, contact_id) {
-            /* we could optimize this by loading only the needed fields */
-            if contact.blocked {
-                *ret_blocked = true;
-            } else {
-                ret = contact.origin;
-            }
-        }
-
-        ret
+    pub fn get_origin_by_id(context: &Context, contact_id: u32) -> Result<(Origin, bool)> {
```

Comment:
Could you document this a little?  I think for something that returns a bool, certainly as part of a tuple, this is a must really.

Additionally the function name is now a bit random.  I wonder if it's better to let the caller create a `Contact` and then use `.is_blocked()` and `.origin` (the origin field is pub).

Reply:
can also mark the Contact `#[non_exhaustive]` and make blocked pub too, which probably makes more sense longer term, but I'm fine not yet doing that.